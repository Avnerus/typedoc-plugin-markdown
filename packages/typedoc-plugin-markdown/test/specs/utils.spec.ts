import * as Handlebars from 'handlebars';

import { TestApp } from '../test-app';

describe(`Utils:`, () => {
  let testApp: TestApp;
  beforeAll(() => {
    testApp = new TestApp(['breadcrumbs.ts']);
  });

  describe(`(default options)`, () => {
    beforeAll(() => {
      testApp.bootstrap({});
    });
    test(`should set named anchors to 'false' by default`, () => {
      expect(
        Handlebars.helpers.ifShowNamedAnchors(TestApp.handlebarsOptionsStub),
      ).toBeFalsy();
    });
    test(`should set breadcrumbs to 'true' by default`, () => {
      expect(
        Handlebars.helpers.ifShowBreadcrumbs(TestApp.handlebarsOptionsStub),
      ).toBeTruthy();
    });
    test(`should set project name to 'true' by default`, () => {
      expect(
        Handlebars.helpers.ifShowProjectName(TestApp.handlebarsOptionsStub),
      ).toBeTruthy();
    });
    test(`should compile relativeURL helper`, () => {
      const url = testApp.findReflection('Breadcrumbs').url;
      expect(Handlebars.helpers.relativeURL(url)).toMatchSnapshot();
    });
  });

  describe(`(set options)`, () => {
    beforeAll(() => {
      testApp.bootstrap({
        publicPath: 'test-public-path',
        namedAnchors: true,
        hideBreadcrumbs: true,
        hideProjectName: true,
      });
    });
    test(`should set named anchors to 'true'`, () => {
      expect(
        Handlebars.helpers.ifShowNamedAnchors(TestApp.handlebarsOptionsStub),
      ).toBeTruthy();
    });
    test(`should set breadcrumbs to 'false'`, () => {
      expect(
        Handlebars.helpers.ifShowBreadcrumbs(TestApp.handlebarsOptionsStub),
      ).toBeFalsy();
    });
    test(`should set project name to 'false'`, () => {
      expect(
        Handlebars.helpers.ifShowProjectName(TestApp.handlebarsOptionsStub),
      ).toBeFalsy();
    });
    test(`should compile relativeURL helper with public path`, () => {
      const url = testApp.findReflection('Breadcrumbs').url;
      expect(Handlebars.helpers.relativeURL(url)).toMatchSnapshot();
    });
  });
});
