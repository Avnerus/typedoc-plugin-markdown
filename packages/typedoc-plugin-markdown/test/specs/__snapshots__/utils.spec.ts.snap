// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils: (default options) should compile relativeURL helper 1`] = `"breadcrumbs.md"`;

exports[`Utils: (set options) should compile relativeURL helper with public path 1`] = `"test-public-pathclasses/breadcrumbs.md"`;
